File: front-end/src/app/service-models/app/v1/client/base-report-filter-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { DateFilterTypeSM } from '../../enums/date-filter-type-s-m.enum';

export class BaseReportFilterSM extends CoinManagementServiceModelBase<number> {
    dateFrom!: Date;
    dateTo!: Date;
    searchString!: string;
    dateFilterType!: DateFilterTypeSM;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/BaseReportFilterSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class BaseReportFilterSM : SiffrumPayrollServiceModelBase<int>
    {
        public DateTime DateFrom { get; set; }
        public DateTime DateTo { get; set; }
        public string SearchString { get; set; }
        public DateFilterTypeSM DateFilterType { get; set; }
    }
}
